src/random.cr

```diff
@@ -1,34 +1,179 @@
-lib C
-  RAND_MAX = 2147483647
+class Random
 
-  fun srand(seed : UInt32)
-  fun rand() : Int32
-end
+  class MT19937
+    N = 624
+    M = 397
+    MATRIX_A = 0x9908b0dfu32
+    UPPER_MASK = 0x80000000u32
+    LOWER_MASK = 0x7fffffffu32
 
-C.srand(Intrinsics.read_cycle_counter.to_u32)
+    def initialize(seeds = Array(UInt32).new(4){ Intrinsics.read_cycle_counter.to_u32 })
```

Comment:
For the default `seeds` we could use:

```crystal
StaticArray(UInt32, 4).new { Intrinsics.read_cycle_counter.to_u32 }
```

In that way no extra memory is allocated. What do you think?

It won't compile because later the method `size` is not found for StaticArray, but we can add it (or change the code to say `length`).